### PR TITLE
Add the 'statistics' command to htool

### PR DESCRIPTION
**Pull request by paipeng-g**

### CODE REVIEW

File: examples/htool_update_failure_reasons.h [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  FIRMWARE_UPDATE_INVALID_DESCRIPTOR_VERSION = 23,
+  FIRMWARE_UPDATE_INVALID_RW_KEY_TRANSITION = 24,
+  FIRMWARE_UPDATE_ERROR_MAX = 25,
+};
+
+// List of common failure codes for payload update.
+enum payload_update_failure_reason {
+  PAYLOAD_UPDATE_SUCCESS = 0,  // EC_SUCCESS
+  PAYLOAD_UPDATE_VALIDATE_RUNTIME_FAILURE =
+      1,  // VALIDATION_ERROR_RUNTIME_FAILURE
```
Comment (cjevans-google) [inline examples/htool_update_failure_reasons.h:53]: nit: These comments are referring to the hoth firmware implementation, which isn't visible to htool anyhow; ok to remove them.

File: examples/htool_update_failure_reasons.h [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  PAYLOAD_UPDATE_CALLBACK_FAILED = 35,
+  PAYLOAD_UPDATE_SET_PENDING_MIGRATION_FAILED = 36,
+  PAYLOAD_UPDATE_CONFIRM_INVALID_TIMEOUT = 37,
+  PAYLOAD_UPDATE_CONFIRM_NOT_ENABLED = 38,
+  PAYLOAD_UPDATE_CONFIRM_NO_UPDATE_PAYLOAD = 39,
+  PAYLOAD_UPDATE_CONFIRM_NO_PENDING_PAYLOAD = 40,
+  PAYLOAD_UPDATE_CONFIRM_GNVRAM_ERROR = 41,
+  PAYLOAD_UPDATE_CONFIRM_REVERT_PAYLOAD = 42,
+  PAYLOAD_UPDATE_CONFIRM_NOT_SUPPORTED =
+      43,  // Does not support confirmed payload update
```
Comment (cjevans-google) [inline examples/htool_update_failure_reasons.h:103]: nit: (Same as above; ok to remove this comment)

**Review by cjevans-google — CHANGES_REQUESTED** [review]

File: examples/htool_statistics.c [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+    fprintf(stderr, "HOTH_STATISTICS error code: %d\n", ret);
+    return -1;
+  }
+
+  printf("Valid statistics words: %u\n", stat.valid_words);
+  uint32_t flags = stat.hoth_reset_flags;
+  printf("Reset flags: 0x%08x\n", flags);
+  const char kIndent[] = "             %s\n";
+  // clang-format off
+  if (flags & kResetFlagOther)      printf(kIndent, "ResetFlagOther");
```
[the remark this answers is not in the document]
Comment (paipeng-g) [reply examples/htool_statistics.c:196]: It should only print "Reset flags: " once

File: examples/htool_update_failure_reasons.h (hunk 1 above)
Comment (paipeng-g) [reply examples/htool_update_failure_reasons.h:53]: Do you mean to remove all the comments?

Comment (paipeng-g) [reply examples/htool_update_failure_reasons.h]: Removed all comments

**Review by cjevans-google — APPROVED** [review]

**Review by cjevans-google — CHANGES_REQUESTED** [review]
Would you mind squashing these three changes into one?  (I can do so via github, but I'd prefer not to because it makes the authorship strange.)

**Comment by paipeng-g** [comment]
How to merge the changes into one? Do I need to create another fork and pull request?

**Comment by cjevans-google** [comment]
You can squash the changes locally (typically with `git rebase --interactive`), and then force push to this branch in github (`git push --force $branch`).

**Review by cjevans-google — APPROVED** [review]

**Pull request merged**